src/test/webapp/app/job/my-positions/my-positions-page.component.spec.ts

```diff
@@ -0,0 +1,292 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { beforeEach, describe, it, expect, vi, Mocked } from 'vitest';
+import { of, throwError } from 'rxjs';
+import { Router } from '@angular/router';
+import { TranslateModule } from '@ngx-translate/core';
+
+import { MyPositionsPageComponent } from 'app/job/my-positions/my-positions-page.component';
+import { ToastService } from 'app/service/toast-service';
+import { JobResourceApiService } from 'app/generated/api/jobResourceApi.service';
+import { AccountService } from 'app/core/auth/account.service';
+import { provideFontAwesomeTesting } from 'src/test/webapp/util/fontawesome.testing';
+import { provideTranslateMock } from 'src/test/webapp/util/translate.mock';
+
+describe('MyPositionsPageComponent', () => {
+  let fixture: ComponentFixture<MyPositionsPageComponent>;
+  let component: MyPositionsPageComponent;
+
+  let jobService: Mocked<JobResourceApiService>;
+  let accountService: Mocked<AccountService>;
+  let toastService: Mocked<ToastService>;
+  let router: Mocked<Router>;
+
+  beforeEach(async () => {
+    jobService = {
+      getAllJobNamesByProfessor: vi.fn().mockReturnValue(of(['A', 'B'])),
+      getJobsByProfessor: vi.fn().mockReturnValue(
+        of({
+          content: [{ jobId: '1', title: 'Job A', state: 'DRAFT' }],
+          totalElements: 1,
+        }),
+      ),
+      deleteJob: vi.fn().mockReturnValue(of({})),
+      changeJobState: vi.fn().mockReturnValue(of({})),
+    } as unknown as Mocked<JobResourceApiService>;
+
+    accountService = {
+      loadedUser: vi.fn().mockReturnValue({ id: 'u1', name: 'User' }),
+    } as unknown as Mocked<AccountService>;
+
+    toastService = {
+      showErrorKey: vi.fn(),
+      showSuccess: vi.fn(),
+      showError: vi.fn(),
+    } as unknown as Mocked<ToastService>;
+
+    router = {
+      navigate: vi.fn(),
+    } as unknown as Mocked<Router>;
+
+    await TestBed.configureTestingModule({
+      imports: [MyPositionsPageComponent, TranslateModule.forRoot()],
+      providers: [
+        { provide: JobResourceApiService, useValue: jobService },
+        { provide: AccountService, useValue: accountService },
+        { provide: ToastService, useValue: toastService },
+        { provide: Router, useValue: router },
+        provideFontAwesomeTesting(),
+        provideTranslateMock(),
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(MyPositionsPageComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should load all job names on init', async () => {
+    await component.loadAllJobNames();
+    expect(jobService.getAllJobNamesByProfessor).toHaveBeenCalled();
+    expect(component.allJobNames()).toEqual(['A', 'B']);
+  });
+
```

Comment:
```suggestion

  it('should assign templates to correct columns', () => {
    fixture.detectChanges();
    
    const columns = component.columns();
    
    expect(columns.find(c => c.field === 'actions')?.template).toBeTruthy();
    expect(columns.find(c => c.field === 'state')?.template).toBeTruthy();
    
    expect(columns.find(c => c.field === 'avatar')?.template).toBeUndefined();
    expect(columns.find(c => c.field === 'professorName')?.template).toBeUndefined();
    expect(columns.find(c => c.field === 'title')?.template).toBeUndefined();
    expect(columns.find(c => c.field === 'startDate')?.template).toBeUndefined();
    expect(columns.find(c => c.field === 'createdAt')?.template).toBeUndefined();
    expect(columns.find(c => c.field === 'lastModifiedAt')?.template).toBeUndefined();
  });

```

Test all column template assignments to catch accidental template changes during refactoring.